### ISSUE TITLE
getScanResults: Neither success nor error callback ever fires

### ISSUE THREAD

**Issue by jorisw** [parsonsmatt/WifiWizard#48]
I can't get WifiWizard.getScanResults() to fire any of its callbacks. I have a scan running, and once the scan runs, I set an interval to retrieve results every 3s. However this never happens, while I can see that the interval does fire.

I am testing with Cordova 5.1.1 and version 0.2.9 (release branch) of your plug-in, on Android 4.1.1.

My code:

```
		if(typeof WifiWizard !== 'undefined'){
			
			WifiWizard.isWifiEnabled(function(wifiIsEnabled){
				
				if(!wifiIsEnabled){
					
					console.log('Wifi is off');
					
				}else{
					
					WifiWizard.listNetworks(function(previouslySelectedNetworks){
						
						if(previouslySelectedNetworks.length>0){
							
							console.log("A network was previously selected but not connected. Inform user why we're presenting the network list now.");
							
							$scope.previouslySelectedNetworkExpired = true;
						}											
	
						WifiWizard.startScan(function(){
							
							console.log("Scanning for networks...");
							
							$scope.scanResultsInterval = setInterval(function(){

								console.log("Scan interval running.");
								
								WifiWizard.getScanResults(function(availableNetworks){
									
									$scope.networks = availableNetworks;
									
									console.log("Updating network list with results." , availableNetworks)
									
								}, function(){
									
									console.log("Failed to get scan results.");
								});
								
							}, 3000);
								
						},function(){
							
							console.log('Failed to start WiFi scan.');
						});
					
						
					}, function(){
						
						console.log('Failed to get a listing of previously selected networks.');
					});
				}
			},function(){
				
			});
		}else{
			
			console.log("No WifiWizard plugin loaded.");
		}
```

**Comment by hoerresb** [comment]
I copy pasted your code and it works... I think. The only difference is I have Android 5.0.1 , but I don't think that should matter. Screen shot of console logs below.
![screenshot 2015-10-19 16 01 58](https://cloud.githubusercontent.com/assets/5693262/10591326/3b04e228-767b-11e5-99f1-5ef4a9a170a5.png)

**Comment by jorisw** [comment]
Thanks for testing my code. Perhaps the developer (or other users) can tell us if the plugin has been tested on Android 4.x. FWIW, the device is a Huawei Ascend Y300.

**Comment by hoerresb** [comment]
Ok. I think I may have figured out the issue. For api version 16 which is what Android 4.1 is using scan results does not include a timestamp field. We look for the timestamp in the code but to not do any checking to see if it exists. So, this is probably causing an exception. I will make a fix and publish it later this week. 

In the meantime, if you want to get this to work I would remove the Wifi wizard plugin and add a new one from a specific branch. 

cordova plugin add https://github.com/parsonsmatt/WifiWizard.git#fix/api_v_16

Let me know if this works.

**Comment by jorisw** [comment]
Thank you for investigating. However removing the plugin and installing your fix did not change the behavior of the plugin on my device. I even brought down the compiling target down to Android SDK 16 to make sure the device OS and the SDK matched, but no cigar. I still get:

```
Scanning for networks...
Scan interval running.
Scan interval running.
```

**Comment by jorisw** [comment]
FYI, same results on Android 4.0.1.

The reason I'm messing around with these old Androids is that 4.x was (and still is) shipped as an embedded system for stuff like HDMI dongles and other such widgets.

**Comment by hoerresb** [comment]
ok. Thanks for testing this. I will continue to look into this issue. At this time I don't have a definitive solution.

**Comment by jorisw** [comment]
Just tested this plugin: https://github.com/floatinghotpot/cordova-plugin-wifi

...and it works on Android 4.1.1 and 4.0.1. I much prefer your API though.

**Comment by hoerresb** [comment]
I looked at that plugin and the code is very similar. I made some changes to the fix/api_v_16 branch, tested it on an android device version 4.1.2 and using your code . It seems to work. I will update master branch this weekend and update the package on npm. 

To check it this works I would remove the current wifiwizard plugin. ```cordova plugin remove com.pylonproducts.wifiwizard ```

and add adjusted branch ```cordova plugin add https://github.com/parsonsmatt/WifiWizard.git#fix/api_v_16```

let me know if this works. Thanks.
